subscriptions-api-openapi.yaml

```diff
@@ -131,120 +138,341 @@ components:
     SubscriptionRequest:
       properties:
         protocol:
-          type: string
-          enum: ["HTTP", "MQTT3", "MQTT5", "AMQP", "NATS", "KAFKA"]
-          description: REQUIRED. Identifier of a delivery protocol.
-          example: "HTTP"
-        protocolsettings:
-          type: object
-          anyOf:
-            - $ref: "#/components/schemas/HTTPSettings"
-            - $ref: "#/components/schemas/MQTTSettings"
-            - $ref: "#/components/schemas/AMQPSettings"
-            - $ref: "#/components/schemas/NATSSettings"
-            - $ref: "#/components/schemas/ApacheKafkaSettings"
-          description: OPTIONAL. A set of settings specific to the selected delivery protocol provider.
+          $ref: "#/components/schemas/Protocol"
+        protocolSettings:
+          oneOf:
```

Comment:
Should this be "anyOf" since it might be possible that a subset of the list of settings could match more than one?

Reply:
I understand your logic, but anyOf is causing code generators to do weirder stuff than oneOf ;) 

We want one of the types to match and the choice is indeed mutually exclusive. Yes, there are odd JSON Schema semantics where oneOf might be violated if the concrete object matches more than one definition. 

I don't know how to fix that other than using the same trick as with the filters.